pkg/canary/reader/reader.go

```diff
@@ -349,7 +349,16 @@ func (r *Reader) run() {
 		// or times out based on the above SetReadDeadline call.
 		err := unmarshal.ReadTailResponseJSON(tailResponse, r.conn)
 		if err != nil {
-			fmt.Fprintf(r.w, "error reading websocket, will retry in 10 seconds: %s\n", err)
+			var reason string
+			netErr, ok := err.(net.Error)
+			if ok && netErr.Timeout() {
+				reason = "timeout tailing new logs"
+			} else {
+				reason = "error reading websocket"
+			}
```

Comment:
You could simplify this a little (this is also a bit more _idiomatic_ Go):

```suggestion
			reason := "error reading websocket"
			
			if e, ok := err.(net.Error); ok && e.Timeout() {
				reason = "timeout tailing new logs"
			}
```

Reply:
I think we should also include the timeout interval in the message.
It's not configurable, so folks will have to examine the source to find it otherwise